plugins/flashrom/fu-flashrom-device.c

```diff
@@ -10,39 +10,68 @@
 
 #include <libflashrom.h>
 
-struct _FuFlashromDevice {
-	FuDevice			 parent_instance;
+typedef struct {
+	FuFlashromDevice		 parent_instance;
+	gchar*				 programmer_name;
+	gchar*				 programmer_args;
 	gsize				 flash_size;
 	struct flashrom_flashctx	*flashctx;
-	struct flashrom_layout		*layout;
 	struct flashrom_programmer	*flashprog;
-};
+} FuFlashromDevicePrivate;
 
-G_DEFINE_TYPE (FuFlashromDevice, fu_flashrom_device, FU_TYPE_DEVICE)
+G_DEFINE_TYPE_WITH_PRIVATE (FuFlashromDevice, fu_flashrom_device, FU_TYPE_DEVICE)
+
+#define GET_PRIVATE(o) (fu_flashrom_device_get_instance_private (o))
+
+void
+fu_flashrom_device_set_programmer_name (FuFlashromDevice *self, const gchar *name)
+{
+	FuFlashromDevicePrivate *priv = GET_PRIVATE (self);
+	g_free (priv->programmer_name);
+	priv->programmer_name = g_strdup (name);
+}
+
+gchar *
+fu_flashrom_device_get_programmer_name (FuFlashromDevice *self)
+{
+	FuFlashromDevicePrivate *priv = GET_PRIVATE (self);
+	return priv->programmer_name;
+}
+
+void
+fu_flashrom_device_set_programmer_args (FuFlashromDevice *self, const gchar *args)
+{
+	FuFlashromDevicePrivate *priv = GET_PRIVATE (self);
+	g_free (priv->programmer_args);
+	priv->programmer_args = g_strdup (args);
+}
+
+gsize
+fu_flashrom_device_get_flash_size (FuFlashromDevice *self)
+{
+	FuFlashromDevicePrivate *priv = GET_PRIVATE (self);
+	return priv->flash_size;
+}
+
+struct flashrom_flashctx *
+fu_flashrom_device_get_flashctx (FuFlashromDevice *self)
```

Comment:
maybe you want a `g_return_val_if_fail` on this function

Reply:
Done. Thank you!